{"num_games": 100, "student_agent_wins": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], "random_agent_wins": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "student_agent_win_count": 99, "random_agent_win_count": 1, "student_agent_win_percentage": 0.99, "random_agent_win_percentage": 0.01, "student_agent_time_taken": 36.568612244129184, "random_agent_time_taken": 0.002212417125701904, "run_info": [{"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.260342836380005, "random_agent_time": 0.0003063678741455078, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 31, "random_agent_score": 5, "student_agent_time": 31.046130180358887, "random_agent_time": 0.0018835067749023438, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 33.482558488845825, "random_agent_time": 0.0010848045349121094, "board_size": 11, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 34.62871694564819, "random_agent_time": 0.002134561538696289, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 37.26987290382385, "random_agent_time": 0.0038192272186279297, "board_size": 11, "swap_players": true, "run_out": "[TLE]:1: Step call took too long, exceeded 2 seconds in step 1.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 706, in all_next_state\n    list_step = self.all_steps(board, adv_pos, my_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 677, in check_valid_step\n    state_queue.append((next_pos, cur_step + 1))\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 38.37798476219177, "random_agent_time": 0.003009796142578125, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 44, "random_agent_score": 1, "student_agent_time": 40.13958501815796, "random_agent_time": 0.0030274391174316406, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 40.77925491333008, "random_agent_time": 0.00270843505859375, "board_size": 11, "swap_players": false, "run_out": "[TLE]:6: Step call took too long, exceeded 2 seconds in step 6.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 663, in check_valid_step\n    if cur_step == max_step:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 40.939584493637085, "random_agent_time": 0.003126859664916992, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 42.33953881263733, "random_agent_time": 0.003548145294189453, "board_size": 11, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.28267788887024, "random_agent_time": 0.000308990478515625, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 31.03065013885498, "random_agent_time": 0.001712799072265625, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 31.00425386428833, "random_agent_time": 0.0010886192321777344, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 29.725624322891235, "random_agent_time": 0.0003654956817626953, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 38.29147410392761, "random_agent_time": 0.0019986629486083984, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 31.17980718612671, "random_agent_time": 0.0006771087646484375, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 35.637691259384155, "random_agent_time": 0.0017943382263183594, "board_size": 11, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 34.62591767311096, "random_agent_time": 0.0019156932830810547, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 38.680132150650024, "random_agent_time": 0.002692699432373047, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 40.42495059967041, "random_agent_time": 0.0037794113159179688, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 78, "random_agent_score": 3, "student_agent_time": 40.51516103744507, "random_agent_time": 0.003495931625366211, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 35.18846774101257, "random_agent_time": 0.0023813247680664062, "board_size": 9, "swap_players": false, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 2, "student_agent_time": 42.918752670288086, "random_agent_time": 0.0042841434478759766, "board_size": 11, "swap_players": true, "run_out": "[TLE]:5: Step call took too long, exceeded 2 seconds in step 5.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 706, in all_next_state\n    list_step = self.all_steps(board, adv_pos, my_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 672, in check_valid_step\n    if np.array_equal(next_pos, end_pos):\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2449, in array_equal\n    return bool(asarray(a1 == a2).all())\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 32.91515278816223, "random_agent_time": 0.0010290145874023438, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 62, "random_agent_score": 1, "student_agent_time": 47.569798707962036, "random_agent_time": 0.004213094711303711, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 33, "random_agent_score": 3, "student_agent_time": 32.83385348320007, "random_agent_time": 0.001543283462524414, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 98, "random_agent_score": 2, "student_agent_time": 38.900211572647095, "random_agent_time": 0.0020418167114257812, "board_size": 10, "swap_players": false, "run_out": "[TLE]:3: Step call took too long, exceeded 2 seconds in step 3.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 672, in check_valid_step\n    if np.array_equal(next_pos, end_pos):\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 98, "random_agent_score": 1, "student_agent_time": 42.34346389770508, "random_agent_time": 0.003906965255737305, "board_size": 10, "swap_players": true, "run_out": "[TLE]:7: Step call took too long, exceeded 2 seconds in step 7.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 672, in check_valid_step\n    if np.array_equal(next_pos, end_pos):\n  File \"<__array_function__ internals>\", line 179, in array_equal\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 118, "random_agent_score": 3, "student_agent_time": 50.7311749458313, "random_agent_time": 0.00542449951171875, "board_size": 11, "swap_players": true, "run_out": "[TLE]:2: Step call took too long, exceeded 2 seconds in step 2.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:6: Step call took too long, exceeded 2 seconds in step 6.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 708, in all_next_state\n    temp = deepcopy(board)\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/copy.py\", line 153, in deepcopy\n    y = copier(memo)\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 57, "random_agent_score": 5, "student_agent_time": 45.75900959968567, "random_agent_time": 0.003614664077758789, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 1, "student_agent_time": 39.154234409332275, "random_agent_time": 0.003468036651611328, "board_size": 11, "swap_players": true, "run_out": "[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 32.95752930641174, "random_agent_time": 0.0010712146759033203, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 29.203933477401733, "random_agent_time": 0.0007345676422119141, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 33, "random_agent_score": 3, "student_agent_time": 34.62898230552673, "random_agent_time": 0.0021653175354003906, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 38.665860652923584, "random_agent_time": 0.0024824142456054688, "board_size": 11, "swap_players": false, "run_out": "[TLE]:1: Step call took too long, exceeded 2 seconds in step 1.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 706, in all_next_state\n    list_step = self.all_steps(board, adv_pos, my_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 677, in check_valid_step\n    state_queue.append((next_pos, cur_step + 1))\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 665, in check_valid_step\n    for dir, move in enumerate(self.moves):\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 79, "random_agent_score": 2, "student_agent_time": 38.89349007606506, "random_agent_time": 0.003187894821166992, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 31.16849422454834, "random_agent_time": 0.0005810260772705078, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 79, "random_agent_score": 1, "student_agent_time": 42.44577240943909, "random_agent_time": 0.003237485885620117, "board_size": 9, "swap_players": false, "run_out": ""}, {"student_agent_score": 1, "random_agent_score": 119, "student_agent_time": 39.205185890197754, "random_agent_time": 0.002070188522338867, "board_size": 11, "swap_players": false, "run_out": "[TLE]:2: Step call took too long, exceeded 2 seconds in step 2.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2449, in array_equal\n    return bool(asarray(a1 == a2).all())\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:3: Step call took too long, exceeded 2 seconds in step 3.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2446, in array_equal\n    if a1.shape != a2.shape:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:5: Step call took too long, exceeded 2 seconds in step 5.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 706, in all_next_state\n    list_step = self.all_steps(board, adv_pos, my_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 671, in check_valid_step\n    continue\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 31.00879669189453, "random_agent_time": 0.0010650157928466797, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 98, "random_agent_score": 2, "student_agent_time": 31.568344116210938, "random_agent_time": 0.0006787776947021484, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 31.05397057533264, "random_agent_time": 0.0007176399230957031, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 42, "random_agent_score": 17, "student_agent_time": 54.74778938293457, "random_agent_time": 0.005583047866821289, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 79, "random_agent_score": 2, "student_agent_time": 37.11346220970154, "random_agent_time": 0.0024251937866210938, "board_size": 9, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.242754697799683, "random_agent_time": 0.0002930164337158203, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 34.898566484451294, "random_agent_time": 0.0020246505737304688, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 1, "student_agent_time": 37.05784463882446, "random_agent_time": 0.0027441978454589844, "board_size": 11, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 41.0264151096344, "random_agent_time": 0.003984212875366211, "board_size": 11, "swap_players": true, "run_out": "[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2449, in array_equal\n    return bool(asarray(a1 == a2).all())\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:6: Step call took too long, exceeded 2 seconds in step 6.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 706, in all_next_state\n    list_step = self.all_steps(board, adv_pos, my_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 672, in check_valid_step\n    if np.array_equal(next_pos, end_pos):\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2449, in array_equal\n    return bool(asarray(a1 == a2).all())\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 32.87033152580261, "random_agent_time": 0.0015287399291992188, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 46, "random_agent_score": 3, "student_agent_time": 34.66977000236511, "random_agent_time": 0.0017364025115966797, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 29.571735382080078, "random_agent_time": 0.0003790855407714844, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 47, "random_agent_score": 1, "student_agent_time": 38.3792245388031, "random_agent_time": 0.003156423568725586, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 31.085949659347534, "random_agent_time": 0.0007407665252685547, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 32, "random_agent_score": 4, "student_agent_time": 31.04262399673462, "random_agent_time": 0.0016515254974365234, "board_size": 6, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 31.29664444923401, "random_agent_time": 0.0007946491241455078, "board_size": 9, "swap_players": false, "run_out": ""}, {"student_agent_score": 62, "random_agent_score": 1, "student_agent_time": 45.674108266830444, "random_agent_time": 0.004771232604980469, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 46.03007936477661, "random_agent_time": 0.0038216114044189453, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 49.443222761154175, "random_agent_time": 0.005747556686401367, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.241763591766357, "random_agent_time": 0.00028395652770996094, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 31.270263195037842, "random_agent_time": 0.0011501312255859375, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 79, "random_agent_score": 1, "student_agent_time": 44.26676893234253, "random_agent_time": 0.003889322280883789, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 35.01687669754028, "random_agent_time": 0.0020782947540283203, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 37.123865365982056, "random_agent_time": 0.001760721206665039, "board_size": 11, "swap_players": false, "run_out": ""}, {"student_agent_score": 56, "random_agent_score": 8, "student_agent_time": 42.102374792099, "random_agent_time": 0.0028166770935058594, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 79, "random_agent_score": 1, "student_agent_time": 42.63171744346619, "random_agent_time": 0.003549337387084961, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 34.70650672912598, "random_agent_time": 0.0014750957489013672, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 24, "random_agent_score": 12, "student_agent_time": 40.06909966468811, "random_agent_time": 0.0027539730072021484, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 99, "random_agent_score": 1, "student_agent_time": 38.7820680141449, "random_agent_time": 0.0029747486114501953, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 2, "student_agent_time": 31.371646642684937, "random_agent_time": 0.0016422271728515625, "board_size": 11, "swap_players": true, "run_out": ""}, {"student_agent_score": 47, "random_agent_score": 1, "student_agent_time": 31.074116468429565, "random_agent_time": 0.0006124973297119141, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 31.00744652748108, "random_agent_time": 0.0011658668518066406, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 97, "random_agent_score": 3, "student_agent_time": 35.190109729766846, "random_agent_time": 0.0018396377563476562, "board_size": 10, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 40.201176166534424, "random_agent_time": 0.0035114288330078125, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 37.17802333831787, "random_agent_time": 0.002104520797729492, "board_size": 11, "swap_players": false, "run_out": "[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.232918977737427, "random_agent_time": 0.00029587745666503906, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 31.20758318901062, "random_agent_time": 0.0010747909545898438, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 1, "student_agent_time": 52.68672776222229, "random_agent_time": 0.0046579837799072266, "board_size": 11, "swap_players": false, "run_out": "[TLE]:5: Step call took too long, exceeded 2 seconds in step 5.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\n  File \"<__array_function__ internals>\", line 181, in array_equal\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:10: Step call took too long, exceeded 2 seconds in step 10.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\n  File \"<__array_function__ internals>\", line 180, in array_equal\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/site-packages/numpy/core/numeric.py\", line 2449, in array_equal\n    return bool(asarray(a1 == a2).all())\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:11: Step call took too long, exceeded 2 seconds in step 11.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 671, in check_valid_step\n    continue\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 58, "random_agent_score": 6, "student_agent_time": 34.76510286331177, "random_agent_time": 0.0016307830810546875, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 98, "random_agent_score": 2, "student_agent_time": 46.52092981338501, "random_agent_time": 0.004191160202026367, "board_size": 10, "swap_players": false, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 38.46494698524475, "random_agent_time": 0.0023081302642822266, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 31.115752458572388, "random_agent_time": 0.0013282299041748047, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 31.04574418067932, "random_agent_time": 0.0006628036499023438, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 48, "random_agent_score": 1, "student_agent_time": 32.86002445220947, "random_agent_time": 0.0013289451599121094, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 47, "random_agent_score": 1, "student_agent_time": 40.18762469291687, "random_agent_time": 0.002333402633666992, "board_size": 7, "swap_players": false, "run_out": ""}, {"student_agent_score": 62, "random_agent_score": 2, "student_agent_time": 34.81178283691406, "random_agent_time": 0.0018312931060791016, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 63, "random_agent_score": 1, "student_agent_time": 31.147611379623413, "random_agent_time": 0.0006194114685058594, "board_size": 8, "swap_players": false, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 35.36784648895264, "random_agent_time": 0.002712249755859375, "board_size": 11, "swap_players": true, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.237324714660645, "random_agent_time": 0.00032067298889160156, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 29.204057455062866, "random_agent_time": 0.0006525516510009766, "board_size": 9, "swap_players": true, "run_out": ""}, {"student_agent_score": 119, "random_agent_score": 2, "student_agent_time": 39.11965847015381, "random_agent_time": 0.002642393112182617, "board_size": 11, "swap_players": false, "run_out": "[TLE]:4: Step call took too long, exceeded 2 seconds in step 4.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n[TLE]:5: Step call took too long, exceeded 2 seconds in step 5.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 45, "random_agent_score": 4, "student_agent_time": 34.70824670791626, "random_agent_time": 0.0017364025115966797, "board_size": 7, "swap_players": true, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.240238904953003, "random_agent_time": 0.0003237724304199219, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 62, "random_agent_score": 1, "student_agent_time": 38.447038412094116, "random_agent_time": 0.0027008056640625, "board_size": 8, "swap_players": true, "run_out": ""}, {"student_agent_score": 80, "random_agent_score": 1, "student_agent_time": 46.258559226989746, "random_agent_time": 0.005422115325927734, "board_size": 9, "swap_players": false, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 31.05850338935852, "random_agent_time": 0.0013959407806396484, "board_size": 11, "swap_players": true, "run_out": ""}, {"student_agent_score": 120, "random_agent_score": 1, "student_agent_time": 31.225627899169922, "random_agent_time": 0.0012843608856201172, "board_size": 11, "swap_players": true, "run_out": ""}, {"student_agent_score": 35, "random_agent_score": 1, "student_agent_time": 29.24215078353882, "random_agent_time": 0.0003211498260498047, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 97, "random_agent_score": 1, "student_agent_time": 39.11467623710632, "random_agent_time": 0.002537250518798828, "board_size": 10, "swap_players": false, "run_out": "[TLE]:2: Step call took too long, exceeded 2 seconds in step 2.\nAn exception raised. The traceback is as follows:\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 54, in time_limit\n    yield\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 255, in run_agent_step\n    next_pos, dir = cur_player.step(\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 225, in step\n    node = self.uct(self.root_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 310, in uct\n    return self.uct(rtn_node)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 293, in uct\n    list_new_board, list_new_pos, list_new_dir = self.all_next_state(node.board, node.my_pos, node.adv_pos, node.my_turn)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 697, in all_next_state\n    list_step = self.all_steps(board, my_pos, adv_pos)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 688, in all_steps\n    if self.check_valid_step(board, np.array(my_pos), np.array([i,j]), k, adv_pos):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/agents/student_agent.py\", line 670, in check_valid_step\n    if np.array_equal(next_pos, adv_pos) or tuple(next_pos) in visited:\nKeyboardInterrupt\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/world.py\", line 292, in step\n    next_pos, dir = self.run_agent_step(\n  File \"/home/ml/users/ksinha4/.conda/envs/comp424/lib/python3.9/contextlib.py\", line 137, in __exit__\n    self.gen.throw(typ, value, traceback)\n  File \"/home/ml/users/ksinha4/mlp/comp424_final_project/submissions/comp424mcgill/final-project-joystick-fighters/timeout.py\", line 56, in time_limit\n    raise TimeoutException(\"Timed out for operation {}\".format(msg))\ntimeout.TimeoutException: Timed out for operation Max allowed : 2 seconds\n\nExecute Random Walk!\n"}, {"student_agent_score": 18, "random_agent_score": 1, "student_agent_time": 43.6386284828186, "random_agent_time": 0.003821134567260742, "board_size": 6, "swap_players": false, "run_out": ""}, {"student_agent_score": 47, "random_agent_score": 2, "student_agent_time": 43.79118275642395, "random_agent_time": 0.003740072250366211, "board_size": 7, "swap_players": true, "run_out": ""}]}